common/datavalues/src/arrays/kernels/take.rs

```diff
@@ -141,9 +148,12 @@ pub unsafe fn take_no_null_primitive_opt_iter_unchecked<
 ) -> Arc<PrimitiveArray<T>> {
     let array_values = arr.values();
 
-    let iter = indices
-        .into_iter()
-        .map(|opt_idx| opt_idx.map(|idx| *array_values.get_unchecked(idx)));
+    let iter = indices.into_iter().map(|opt_idx| {
+        opt_idx.map(|idx| {
+            let _ = &array_values;
```

Comment:
Why rust 2021 turns to be more complex.

Reply:
:sob:

Reply:
> Why rust 2021 turns to be more complicated.

I don't think it becomes more complicated. 

> This is a conservative analysis: in many cases, these dummy lets can be safely removed and your program will work fine.

> Starting in Rust 2021, closures captures are more precise. Typically they will only capture the fields they use (in some cases, they might capture more than just what they use, see the Rust reference for full details).